spl/src/associated_token.rs

```diff
@@ -8,6 +8,28 @@ pub use spl_associated_token_account::{
     get_associated_token_address, get_associated_token_address_with_program_id, ID,
 };
 
+pub fn create_slim_stack_frame<'info>(
+    payer: &AccountInfo<'info>,
+    associated_token: &AccountInfo<'info>,
+    authority: &AccountInfo<'info>,
+    mint: &AccountInfo<'info>,
+    system_program: &AccountInfo<'info>,
+    token_program: &AccountInfo<'info>,
+    associated_token_program: &AccountInfo<'info>,
+) -> Result<()> {
+    create(CpiContext::new(
+        associated_token_program.clone(),
+        Create {
+            payer: payer.clone(),
+            associated_token: associated_token.clone(),
+            authority: authority.clone(),
+            mint: mint.clone(),
+            system_program: system_program.clone(),
+            token_program: token_program.clone(),
+        },
+    ))
+}
```

Comment:
It would be better if we can avoid adding this function and then to only use it once.

A similar issue has been resolved in https://github.com/coral-xyz/anchor/pull/2913 by inlining variables, maybe that works here too? If not, maybe creating closures like in https://github.com/coral-xyz/anchor/pull/2939 helps?

Reply:
I inlined variables just as in #2913.